### ISSUE TITLE
Using pre-trained PMTransformer predict COFs structure information

### ISSUE THREAD

**Issue by Rider99** [hspark1212/MOFTransformer#158]
Hi @Yeonghun1675,
Your work is very interesting and powerful. 
I am currently interested in using the PMtransformer to predict structural information about COFs structures' information, such as "topology" and "building block classification". However, I’m unsure what the “downstream” for these tasks, or how to utilize the pretrained model alone since the documentation(https://hspark1212.github.io/MOFTransformer/index.html) I found only outlines steps for finetuning. 
Could you assist me with this?
Thanks for your help.

**Comment by Yeonghun1675** [comment]
Hi @Rider99,

Thank you for your interest in our pmtransformer. Currently, our model does not directly provide predictions of MOC and MTP. However, you can use pre-trained MOCs and MTPs in PMTransformer in the following ways

```python
import os
import copy
import json
import torch
import moftransformer
from moftransformer import __root_dir__

# copy config file
config = moftransformer.config.config()
config = copy.deepcopy(config)

config['downstream'] = 'mtp'
config['loss_names'] = {'ggm': 0,
        'mpp': 0,
        'mtp': 1,
        'vfp': 0,
        'moc': 1,
        'bbc': 0,
        'classification': 0,
        'regression': 0
}
# include your configuration such as batch_size!

model = moftransformer.modules.module.Module(config)
dm = moftransformer.datamodules.datamodule.Datamodule(config)

dm.setup()

with open(os.path.join(__root_dir__, 'assets/topology.json')) as f:
    topo = list(json.load(f).keys())

loader = (iter(dm.train_dataloader()))  # you also can use test / validation dataloader
for batch in loader:
    output = model(batch)
    
    # calculate mtp
    softmax = torch.nn.Softmax(dim=1)
    mtp = softmax(output['mtp_logits'])
    mtp = torch.argmax(mtp, dim=1)
    mtp = [topo[int(t)] for t in mtp]
    
    # calculate moc
    sigmoid = torch.nn.Sigmoid()
    moc = sigmoid(output['moc_logits'])
    moc = moc > 0.5
    moc = moc.reshape(your_batchsize, -1)
    atom_index = batch['uni_idx']
    
    metal_types = []
    for indices, labels in zip(batch['uni_idx'], moc):
        for idx, label in zip(indices, labels):
            if label:
                 metal_types.append(idx)
```

You can do this by creating `train_moc.json` and `train_mtp.json` where root_dataset is located (each label should be an int) and running them. Please check the MOC in case the labels are reversed.

**Comment by Rider99** [comment]
Thank you so much!
